src/etc/htmldocck.py

```diff
@@ -350,7 +350,7 @@ def resolve_path(self, path):
     def get_absolute_path(self, path):
         return os.path.join(self.root, path)
 
-    def get_file(self, path):
+    def get_file(self, path, need_content):
```

Comment:
Changes in this file were needed because `tests/rustdoc/cross-crate-info/working-dir-examples/q.rs` use the `@has` command on the metadata file generated by `--scrape-examples`. However, with my changes, some of its content is not utf8 compatible anymore, breaking the test.

Now, it only read the content if we actually need to match the content of this file.